### ISSUE TITLE
Feature request: support different overlay repository URLs w/o warnings

### ISSUE THREAD

**Issue by CraigFe** [ocamllabs/opam-monorepo#168]
The tool currently hard-codes the path to the `opam-overlays` repository ([here](https://github.com/ocamllabs/opam-monorepo/blob/master/lib/config.ml#L30)), and emits a warning if that particular URL isn't one of the active repositories in the current Opam switch:

```
opam-monorepo: [WARNING] The dune-universe opam-repository isn't set in the current switch. It contains dune ports for some opam packages. Note that opam monorepo lock will fail if not all of the project dependencies use dune as their build system. Adding this opam-repository to your current switch will help with that. If you wish to do so, run the following command:
```

I'm currently using `opam monorepo` with [my own set of overlays](https://github.com/CraigFe/tezos-monorepo-overlays), and so want to avoid pulling packages from the default repository. It would be nice to avoid warnings in this case.

Some things that might help:

- require the switch to have a repository _named_ `dune-universe` (which can point to any URL);
- only emit this warning if the lock fails due to packages not building with Dune.

**Comment by NathanReb** [comment]
That's a good point and that kind of workflow is the very reason why I decided to let the user chose whether they want to use this repo or not, it used to be hardcoded.

I like your suggestion that we only emit the warning when the solving fails because some dependencies don't build with dune AND the opam-overlays repo isn't set. Would that be enough for you?

I guess that a nice side effect of this could be that we don't load the `RepositoryState` everytime anymore which, if I remember correctly, would also benefit you since you had issues with this step slowing down opam-monorepo's startup.

**Comment by CraigFe** [comment]
That sounds great :-) If it avoids needing to load global Opam state, that's a very nice bonus.

**Comment by NathanReb** [comment]
This has been fixed by #210 !

Closing!
